### PR TITLE
Add HealthDataRecord3 to data warehouse

### PR DESCRIPTION
**Pull request by eric-wu**

### CODE REVIEW

File: src/main/resources/sql/bridge_healthdatarecord3.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS bridge_healthdatarecord3_<dateSuffix> (
+    id              CHAR(36)        ENCODE RAW          NOT NULL,
+    healthCode      CHAR(36)        ENCODE RAW          NOT NULL,
+    createdOn       BIGINT          ENCODE DELTA32k             ,
+    studyId         VARCHAR(50)     ENCODE BYTEDICT             ,
+    uploadId        CHAR(36)        ENCODE RAW          NOT NULL,
+    uploadDate      DATE            ENCODE RAW                  ,
+    schemaId        VARCHAR(250)    ENCODE LZO                  ,
```
Comment (DwayneJengSage) [inline src/main/resources/sql/bridge_healthdatarecord3.sql]: There are ~100 schemaIds, and this set grows very slowly. Would it be worthwhile to use a bytedict instead?

File: src/main/resources/sql/bridge_healthdatarecord3.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    schemaRevision  INT             ENCODE DELTA                ,
+    data            VARCHAR(10000)  ENCODE LZO                  ,
+    metaData        VARCHAR(5000)   ENCODE LZO                  ,
+    version         INT             ENCODE DELTA
+) SORTKEY (studyId, uploadDate, schemaId, healthCode, uploadId);
```
Comment (DwayneJengSage) [inline src/main/resources/sql/bridge_healthdatarecord3.sql]: Is sortkey the same as an index? Would it be useful to have a key on id (record ID)?

**Comment by DwayneJengSage** [comment]
What about userSharingScope?

File: src/main/resources/sql/bridge_healthdatarecord3.sql (hunk 1 above)
Comment (eric-wu) [inline src/main/resources/sql/bridge_healthdatarecord3.sql]: According to document, this stores at most 256 (the capacity of 1 byte) unique values compressed.  It's best for enum-type data.  I think we will soon grow out of 256 schemas once we introduce more studies.

File: src/main/resources/sql/bridge_healthdatarecord3.sql (hunk 2 above)
Comment (eric-wu) [inline src/main/resources/sql/bridge_healthdatarecord3.sql]: They are sorted as opposed to e.g. hashed.  So unless we query a lot on record ID, there is no benefit in including it in the sort key list.

**Pull request merged**